src/io/writer_public.cpp

```diff
@@ -7,14 +7,22 @@
 namespace copc
 {
 
-Writer::Writer(std::ostream &out_stream, LasConfig const &config, int span, const std::string &wkt)
+Writer::Writer(std::ostream *out_stream, LasConfig const &config, int span, const std::string &wkt)
 {
     auto header = HeaderFromConfig(config);
     this->file_ = std::make_shared<CopcFile>(header, span, wkt, config.extra_bytes);
     this->hierarchy_ = std::make_shared<Internal::Hierarchy>();
     this->writer_ = std::make_unique<Internal::WriterInternal>(out_stream, this->file_, this->hierarchy_);
 }
 
+Writer::Writer(const std::string &file_path, LasConfig const &config, int span, const std::string &wkt)
+{
+    auto header = HeaderFromConfig(config);
+    this->file_ = std::make_shared<CopcFile>(header, span, wkt, config.extra_bytes);
```

Comment:
Refactor all this code into an InitWriter(ostream stream) function. Then in the NamedWriter constructor, create the stream and call InitWriter